lib/ansible/utils/display.py

```diff
@@ -619,14 +654,21 @@ def do_var_prompt(self, varname, private=True, prompt=None, encrypt=None, confir
             result = wrap_var(result)
         return result
 
-    def _set_column_width(self):
+    def _set_column_width(self) -> None:
         if os.isatty(1):
             tty_size = unpack('HHHH', fcntl.ioctl(1, termios.TIOCGWINSZ, pack('HHHH', 0, 0, 0, 0)))[1]
         else:
             tty_size = 0
         self.columns = max(79, tty_size - 1)
 
-    def prompt_until(self, msg, private=False, seconds=None, interrupt_input=None, complete_input=None):
+    def prompt_until(
+        self,
+        msg: str,
+        private: bool = False,
+        seconds: int | None = None,
+        interrupt_input: t.Iterable[bytes] | None = None,
+        complete_input: t.Iterable[bytes] | None = None,
```

Comment:
`typing.Iterable` is deprecated. Use `collections.abc.Iterable[bytes]` instead.

Reply:
Also, it's tricky with iterables of `bytes` and `str`. For example, `str` is `Iterable[str]`. And an empty string `''` is an `Iterable[str]` too. So you may want to either list explicit possibilities or add a runtime assertion in the code base checking for this not being a string.

Reply:
Following https://github.com/ansible/ansible/pull/81400/files/65fcbf40b3ce6980ccc1c3147dbadd6ef3b64522#r1284875220, these may be `collections.abc.Container[bytes] | None` but that needs verification.